Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -274,51 +252,29 @@ void TextureCacheBase::ScaleTextureCacheEntryTo(TextureCacheBase::TCacheEntryBas
     return;
   }
 
-  TextureCacheBase::TCacheEntryConfig newconfig;
+  TextureConfig newconfig;
   newconfig.width = new_width;
   newconfig.height = new_height;
-  newconfig.layers = (*entry)->config.layers;
+  newconfig.layers = entry->Layers();
   newconfig.rendertarget = true;
 
-  TCacheEntryBase* newentry = AllocateTexture(newconfig);
-  if (newentry)
+  auto new_texture = AllocateTexture(newconfig);
+  if (new_texture)
   {
-    newentry->SetGeneralParameters((*entry)->addr, (*entry)->size_in_bytes, (*entry)->format);
-    newentry->SetDimensions((*entry)->native_width, (*entry)->native_height, 1);
-    newentry->SetHashes((*entry)->base_hash, (*entry)->hash);
-    newentry->frameCount = frameCount;
-    newentry->is_efb_copy = (*entry)->is_efb_copy;
-    MathUtil::Rectangle<int> srcrect, dstrect;
-    srcrect.left = 0;
-    srcrect.top = 0;
-    srcrect.right = (*entry)->config.width;
-    srcrect.bottom = (*entry)->config.height;
-    dstrect.left = 0;
-    dstrect.top = 0;
-    dstrect.right = new_width;
-    dstrect.bottom = new_height;
-    newentry->CopyRectangleFromTexture(*entry, srcrect, dstrect);
-
-    // Keep track of the pointer for textures_by_hash
-    if ((*entry)->textures_by_hash_iter != textures_by_hash.end())
-    {
-      newentry->textures_by_hash_iter = textures_by_hash.emplace((*entry)->hash, newentry);
-    }
+    auto srcrect = entry->texture->config.Rect();
+    auto dstrect = new_texture->config.Rect();
 
-    InvalidateTexture(GetTexCacheIter(*entry));
-
-    *entry = newentry;
-    textures_by_address.emplace((*entry)->addr, *entry);
+    new_texture->CopyRectangleFromTexture(entry->texture.get(), srcrect, dstrect);
+    entry->texture.swap(new_texture);
```

Comment:
I think with the current code, this is indeed a rare event.

But I could see future hybrid xfb putting pressure on this code (and the current buggy version of the hybrid xfb pr actually shoves a texture through here every frame, for games with tiled xfb copies).

You could argue that because it's rare, there is a good chance the texture is of a weird size and  won't be reused, but probably idea to pool it anyway.